front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -258,155 +265,318 @@ const createTrainSchedulePayload = async ({
+ * Generate a TimetableItemBase payload from a trainrun.
+ */
+const generateTimetableItemBase = async ({
+  trainrunSections,
+  nodes,
+  trainrun,
+  infraId,
+  dispatch,
+  labels,
+  timetableItem,
+}: {
+  trainrunSections: TrainrunSectionDto[];
+  nodes: NodeDto[];
+  trainrun: TrainrunDto;
+  infraId: number;
+  dispatch: AppDispatch;
+  labels: LabelDto[];
+  timetableItem?: TimetableItemWithTimetableId;
+}): Promise<TimetableItemBase> => {
+  const path = await generatePath(trainrunSections, nodes, infraId, dispatch);
+  const trainrunLabels = extractTrainrunLabels(trainrun, labels);
+  const startDate = calculateStartDate(
+    trainrunSections,
+    timetableItem ? new Date(timetableItem.start_time) : new Date()
+  );
+  const schedule = generateSchedule(trainrunSections, nodes, startDate);
+
+  if (!timetableItem) {
+    return generateDefaultPacedTrainBase(trainrun, trainrunLabels, path, startDate, schedule);
+  }
+  if (isPacedTrain(timetableItem.id)) {
+    return generatePacedTrainBase(
+      trainrun,
+      trainrunLabels,
+      path,
+      startDate,
+      schedule,
+      timetableItem as PacedTrainResultWithPacedTrainId
+    );
+  }
+  return generateTrainScheduleBase(trainrun, trainrunLabels, path, startDate, schedule);
+};
+
+const fetchTimetableItem = async (
+  timetableItemId: TimetableItemId,
+  dispatch: AppDispatch
+): Promise<TimetableItemWithTimetableId> => {
+  if (isPacedTrain(timetableItemId)) {
+    const pacedTrain = await dispatch(
+      osrdEditoastApi.endpoints.getPacedTrainById.initiate({
+        id: formatPacedTrainIdToEditoastTrainId(timetableItemId),
+      })
+    ).unwrap();
+    return { ...pacedTrain, id: timetableItemId };
+  }
+  const trainSchedule = await dispatch(
+    osrdEditoastApi.endpoints.getTrainScheduleById.initiate({
+      id: formatTrainScheduleIdToEditoastTrainId(timetableItemId),
+    })
+  ).unwrap();
+  return { ...trainSchedule, id: timetableItemId };
+};
+
+const createPacedTrain = async (
+  timetableId: number,
+  pacedTrain: PacedTrainBase,
+  dispatch: AppDispatch
+): Promise<PacedTrainResultWithPacedTrainId> => {
+  const newTimetableItems = await dispatch(
+    osrdEditoastApi.endpoints.postTimetableByIdPacedTrains.initiate({
+      id: timetableId,
+      body: [pacedTrain],
+    })
+  ).unwrap();
+  if (!newTimetableItems || newTimetableItems.length === 0) {
+    throw new Error('Failed to create paced train');
+  }
   return {
-    train_name: trainrun.name,
-    labels: compact(trainrunLabels),
-    path: path.flat(),
-    start_time: startDate.toISOString(),
-    schedule,
+    ...newTimetableItems[0],
+    id: formatEditoastTrainIdToPacedTrainId(newTimetableItems[0].id),
   };
 };
 
-const handleUpdateTrainSchedule = async ({
+const updateTimetableItem = async (
+  timetableItemId: TimetableItemId,
+  timetableItem: TimetableItemWithTimetableId,
+  dispatch: AppDispatch
+): Promise<TimetableItemWithTimetableId> => {
+  if (isPacedTrain(timetableItemId)) {
```

Comment:
You can use the new function `isPacedTrainResultWithPacedTrainId` which checks directly the type of the timetableItem, so you can: 
- remove all the cast of the functions
- remove timetableItemId (use timetableItem.id)
```suggestion
  if (isPacedTrainResultWithPacedTrainId(timetableItem.id)) {
```

Reply:
Done in "refactor update to reduce casting"